internal/reconciler/shoot/reconciler_test.go

```diff
@@ -0,0 +1,241 @@
+					},
+					ResourceVersion: "1",
+				},
+				Spec: authenticationv1alpha1.OIDCAuthenticationSpec{
+					IssuerURL:      "https://shoot/issuer",
+					ClientID:       "garden",
+					UsernameClaim:  ptr.To("sub"),
+					UsernamePrefix: ptr.To(fmt.Sprintf("ns:%s:shoot:%s:%s:", shoot.Namespace, shoot.Name, string(shoot.UID))),
+					GroupsClaim:    ptr.To("groups"),
+					GroupsPrefix:   ptr.To(fmt.Sprintf("ns:%s:shoot:%s:%s:", shoot.Namespace, shoot.Name, string(shoot.UID))),
+				},
+			},
+		))
+	})
+
+	It("should do nothing because shoot has no managed issuer annotation", func() {
+		shoot.Annotations = map[string]string{}
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+
+		res, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: shootObjectKey})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		var oidcList authenticationv1alpha1.OpenIDConnectList
+		Expect(fakeClient.List(ctx, &oidcList)).To(Succeed())
+		Expect(oidcList.Items).To(BeEmpty())
+	})
+
+	It("should delete OIDC resource because shoot is not trusted", func() {
+		shoot.Annotations = map[string]string{
+			"authentication.gardener.cloud/issuer":  "managed",
+			"authentication.gardener.cloud/trusted": "false",
+		}
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+		// Create OIDC resource that should be deleted
+		Expect(fakeClient.Create(ctx, oidc)).To(Succeed())
+
+		res, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: shootObjectKey})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		var oidcList authenticationv1alpha1.OpenIDConnectList
+		Expect(fakeClient.List(ctx, &oidcList)).To(Succeed())
+		Expect(oidcList.Items).To(BeEmpty())
+	})
+
+	It("should do nothing when shoot is not trusted and OIDC resource does not exist", func() {
+		shoot.Annotations = map[string]string{
+			"authentication.gardener.cloud/issuer":  "managed",
+			"authentication.gardener.cloud/trusted": "false",
+		}
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+
+		res, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: shootObjectKey})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		var oidcList authenticationv1alpha1.OpenIDConnectList
+		Expect(fakeClient.List(ctx, &oidcList)).To(Succeed())
+		Expect(oidcList.Items).To(BeEmpty())
+	})
+
+	It("should delete OIDC resource because shoot is being deleted", func() {
+		// Adding a finalizer to simulate that the shoot is being deleted and not yet fully deleted to trigger the shoot.DeletionTimestamp check
+		shoot.Finalizers = []string{"some/finalizer"}
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+		// Create OIDC resource that should be deleted
+		Expect(fakeClient.Create(ctx, oidc)).To(Succeed())
+
+		res, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: shootObjectKey})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
```

Comment:
```suggestion
```

Having in mind that the OIDC is pre-created, this reconciliation is not expected to do anything, from this point of view, I think it can be removed, am I right?